src/scene_battle_rpg2k3.cpp

```diff
@@ -2253,11 +2254,55 @@ Scene_Battle_Rpg2k3::BattleActionReturn Scene_Battle_Rpg2k3::ProcessBattleAction
 			auto* weapon_animation_data = action->GetWeaponAnimationData();
 			if (weapon_animation_data) {
 				cba_num_ranged_weapon_move_frames = (weapon_animation_data->ranged_speed + 1) * 20;
-				cba_ranged_weapon = std::make_unique<Sprite_Weapon>(actor);
-				cba_ranged_weapon->SetWeaponAnimation(weapon_animation_data->ranged_animation_id + 1);
-				cba_ranged_weapon->SetRanged(true);
-				cba_ranged_weapon->StartAttack(action->GetSourcePose() == lcf::rpg::BattlerAnimation::Pose_AttackLeft);
-				cba_ranged_weapon->Update();
+
+				// The ranged weapon animation targets the original single target
+				// if the weapon has the "Attack All" flag set and the ranged attack
+				// range is set to "Single Enemy"
+				if (action->GetWeaponData()->attack_all) {
+					if (action->GetWeaponData()->ranged_target == lcf::rpg::Item::Target_single) {
+						cba_ranged.emplace_back(*action->GetOriginalSingleTarget(), nullptr);
+					} else {
+						if (action->GetWeaponData()->ranged_target == lcf::rpg::Item::Target_center) {
+							std::vector<Game_Battler*> enemies;
+							Main_Data::game_enemyparty->GetActiveBattlers(enemies);
+							int x = 0;
+							int y = 0;
+							for (Game_Battler* enemy : enemies) {
+								x += enemy->GetBattlePosition().x;
+								y += enemy->GetBattlePosition().y;
+							}
+							if (enemies.size() > 0) {
+								x /= enemies.size();
+								y /= enemies.size();
+							}
+							cba_ranged_center = Point(x, y);
+
+							// This is needed to make the ranged weapon animation appear
+							// even if the real target of the animation is the center
+							cba_ranged.emplace_back(*action->GetTarget(), nullptr);
+						} else {
+							if (action->GetWeaponData()->ranged_target == lcf::rpg::Item::Target_simultaneous) {
+								std::vector<Game_Battler*> battlers;
+								Main_Data::game_enemyparty->GetActiveBattlers(battlers);
+								for (auto& b: battlers) {
+									cba_ranged.emplace_back(*b, nullptr);
+								}
+							}
```

Comment:
Just saw one more code problem:
You have a
```
if () {
} else {
  if () {
  } else {
}
```
here. This can be reduced to multiple ``else if`` instead.

Reply:
Fixed.